soh/src/overlays/actors/ovl_Bg_Toki_Swd/z_bg_toki_swd.c

```diff
@@ -139,7 +133,8 @@ void func_808BAF40(BgTokiSwd* this, GlobalContext* globalCtx) {
             this->actor.parent = NULL;
             BgTokiSwd_SetupAction(this, func_808BB0AC);
         } else {
-            if (Actor_IsFacingPlayer(&this->actor, 0x2000)) {
+            Player* player = GET_PLAYER(globalCtx);
+            if (Actor_IsFacingPlayer(&this->actor, 0x2000) && (gSaveContext.n64ddFlag && player->getItemId == GI_NONE)) {
```

Comment:
looks like this changes (breaks?) vanilla logic

Reply:
Oh snap you're right, Gotta or that added case with !gSaveContext.n64ddFlag

Reply:
I think I've fixed it but testing vanilla behavior right now and gotta sit through a few cutscenes lol.

Reply:
Well I found a bug in vanilla behavior but not with that line. Sheik just completely vanished after the cutscene lol.

Reply:
I think I found the line. I imagine this was supposed to kill Sheik's actor in rando, but it wasn't put behind a check for the n64dd flag. This is the function that's supposed to change Sheik's animation to the crossed arms animation while they are blocking the pedestal.

```C
void func_80B3C9EC(EnXc* this) {
    EnXc_ChangeAnimation(this, &gSheikArmsCrossedIdleAnim, ANIMMODE_LOOP, 0.0f, false);
    this->action = SHEIK_ACTION_BLOCK_PEDESTAL;
    this->drawMode = SHEIK_DRAW_DEFAULT;
    this->unk_30C = 1;
    Actor_Kill(&this->actor);
    return;
}
```

Reply:
Ok, after fixing that all the other vanilla behavior seems to work fine. Next I'll retest the rando behavior and then commit.

Reply:
ah, yes, the quick and dirty actor kill from a "the stream must go on" [commit](https://github.com/briaguya-ai/Shipwright/commit/058a691f50f6184179ea7b98cfea307b88ca4def)...

great catch!